[Feat] #35 cafe info

WooBinHam opened the pull request:
## 카페 상세 페이지
- 카페 상세 Activity에 두가지 버튼을 두고 버튼 클릭시 해당 Fragment로 전환.
- 단, 리뷰 및 평점 Fragment로 넘어갈 때에는 하단에 '리뷰쓰기'버튼이 생겼다가 메뉴 Fragment에서는 사라짐.
- 전체적인 코드 연결을 통해 중복되는 부분 해결해야함. 예를들어 이미지 배열
- 이미지 슬라이드시 dot indicator 연동
- 서버 연결 후 더미 데이터 삭제해야함
- 서버와의 소통 필요

Commit:
[Feat] #35 - 카페 상세 정보

- 임시 commit

Commit:
[Feat] #35 - 카페 상세 정보

- 임시 commit

Commit:
Merge remote-tracking branch 'origin/feature/35-cafe-info' into feature/35-cafe-info

Commit:
[Feat] #35 - 카페 상세 페이지

- Dot Indicator gradle 추가
- DataBinding을 위한 CafeMenu DataClass 추가
- 카페 상세페이지에서 Base가 되는 activity layout 완료
- 메뉴 frag에서 아이템까지는 생성. 그 이후에 더미 데이터로 테스트
- 리뷰 데이터는 아이템 .xml 만들어야

Commit:
[Feat] #35 - 카페 메뉴

- 카페 메뉴 더미 데이터 넣어서 메뉴 Frag 구현 완료
- 아이템 그림자는 constraint의 elevation 사용
- 그림자 효과를 위한 메뉴와 리뷰 아이템을 위한 drawable 배경 추가
- CafeMenu & CafeMenuRes 데이터 클래스 생

Commit:
[Feat] #35 - 카페 상세 페이지

- 리뷰 Fragment 구현 완료
- ViewPager2를 이용해서 상세 페이지 내부 이미지 슬라이드 가능
- 카페 상세 Activity와 두가지 Fragment 연결 완료
- 추가로 '리뷰쓰기'버튼 와리가리 가능
- 서버와 연결하여 데이터 주고받기만 하면 완료

WooBinHam committed:
Merge branch 'develop' into feature/35-cafe-info

YiBeomSeok commented on app/src/main/java/org/cazait/data/model/CafeMenu.kt:
```diff
@@ -0,0 +1,13 @@
+    @SerializedName("price") val price: Int,
+    @SerializedName("imageUrl") val imageUrl: String
+) {
+    fun getStringPrice() = price.toString()
```
이런 방법도 추천
ex) 
```kotlin
val stringPrice: String
get() = price.toString()
```

```kotlin
...
val price = cafeMenu.stringPrice
```

WooBinHam replied:
`binding.tvCafeMenuPrice.text = item.getStringPrice()`
CafeInfoMenuAdapter에서 위와 같이 수정했습니다.

YiBeomSeok commented on app/src/main/java/org/cazait/data/model/response/CafeReviewRes.kt:
```diff
@@ -0,0 +1,17 @@
+    @SerializedName("message") val message: String,
+    @SerializedName("data") val data: Data
+) {
+    data class Data(
```
클래스 밖으로 빼는게 좋을 것 같습니다
작명이 귀찮긴 하지만..

YiBeomSeok commented on app/src/main/java/org/cazait/ui/adapter/CafeImgAdapter.kt:
이 부분 adapter를 ListAdapter로 바꾸는 것 나중에 리펙토링 할 때 고려해보세용

YiBeomSeok commented on app/src/main/java/org/cazait/ui/adapter/CafeInfoMenuAdapter.kt:
```diff
@@ -0,0 +1,51 @@
+        fun bind(item: CafeMenu) {
+//            binding.ivCafeMenu.setImage(item.imageUrl)
+            binding.tvCafeMenu.text = item.name
+            binding.tvCafeMenuPrice.text = item.price.toString()
```
CafeMenu에서 stringPrice 함수가 있던 것 같은데 함수를 사용하거나, 그냥 바인딩할 때 toString 써도 상관 없을 것 같네요

YiBeomSeok commented on app/src/main/res/layout/fragment_cafe_info_menu.xml:
```diff
@@ -0,0 +1,39 @@
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <FrameLayout
```
루트 레이아웃이 프레임레이아웃이어야 하나요?

WooBinHam replied:
꼭 framelayout 일 필요는 없지만 fragment 생성할 때 기본적으로 framelayout이 만들어져 있는 상태라 그 안에 constraint를 넣었습니다.
framelayout을 빼고 constraint를 루트 레이아웃으로 해봐도 문제 없어서 layout 한단계 줄이겠습니다.

YiBeomSeok commented on app/src/main/res/layout/fragment_cafe_info_review.xml:
```diff
@@ -0,0 +1,39 @@
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <FrameLayout
```
.

YiBeomSeok reviewed (approved):
xml파일에서 폰트도 지정해주면 좋겠습니다~

Commit:
[Chore] #35 - 요구사항 수정

- 본인이 작업했던 모든 xml 파일에 대해 폰트 추가
- fragment에서 루트 layout을 constraint로 변경. framelayout 삭제
- CafeInfoMenuAdapter에서 가격 정보 불러오는 방식 변경
- CafeReviewRes의 Data 데이터 클래스 밖으로 방출

Commit:
Merge remote-tracking branch 'origin/feature/35-cafe-info' into feature/35-cafe-info
